src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/AbstractReadThreadingGraph.java

```diff
@@ -594,25 +594,25 @@ int mergeDanglingHeadLegacy(final DanglingChainMergeHelper danglingHeadMergeResu
 
         final List<CigarElement> elements = danglingHeadMergeResult.cigar.getCigarElements();
         final CigarElement firstElement = elements.get(0);
-        Utils.validateArg(firstElement.getOperator() == CigarOperator.M, "The first Cigar element must be an M");
```

Comment:
I would recommend removing the changes to this class since they are holdovers from before the rebase of the dangling head branch.